### PR TITLE
[JENKINS-73554] fix jelly exception

### PR DESCRIPTION
**Pull request by mawinter69**
ensure that we don't get an IndexOutOfBoundsException when the controller has no executors on cloud pages

See [JENKINS-73554](https://issues.jenkins.io/browse/JENKINS-73554).

### Testing done
Manual testing as described in the Jira

### Proposed changelog entries

- fix IndexOutOfBoundsException in cloud management pages when controller has no executors (regression in 2.463).

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: core/src/main/resources/lib/hudson/executors.jelly
Comment (timja) [inline core/src/main/resources/lib/hudson/executors.jelly:146]: don't you want gt 0?
```suggestion
    <j:when test="${!builtInHasExecutors and computers.size() gt 0 and computers.get(0).node == app}">
```

Comment (mawinter69) [reply core/src/main/resources/lib/hudson/executors.jelly:146]: It should be gt 1, otherwise it fails when visiting the built-in executor page when that has 0 executors

Comment (timja) [reply core/src/main/resources/lib/hudson/executors.jelly:146]: Is gt not greater than 0?

Comment (mawinter69) [reply core/src/main/resources/lib/hudson/executors.jelly:146]: It must be `gt 1` otherwise you get an `IndexOutOfBoundsException` in line 158 when visiting the built-in agent with 0 executors. In that case have a singleExecutor and we don't want the `computers` to be empty which would happen with `gt 0`

**Review by timja — APPROVED** [review]

**Review by StefanSpieker — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
